frontend/src/Judge/reducers.js

```diff
@@ -78,6 +95,34 @@ const queue = (state = initialQueueState, action) => {
 const queues = (state = {}, action) => {
   // Proxy all queue actions to the particular queue we want to target
   switch (action.type) {
+    case actions.FETCH_SUBMISSIONS:
+      const {
+        submissions,
+        username
+      } = action.payload
+      // First, gather the submissions by show ID (they _should_ be all the same
+      // ID anyway, but just in case)
+      const submissionsByShowId = submissions.reduce(
```

Comment:
In what scenario would they not all have the same ID?

Reply:
I just felt a bit odd using `const showId = submissions[0].show.id` and assuming they're all the same. If that's okay though, then I can remove this bit.